packages/transport/src/utils/readMessageBuffer.ts

```diff
@@ -48,8 +48,18 @@ export const readMessageBuffer = () => {
         });
     };
 
+    const close = (path: string) => {
+        if (readRequests[path]) {
+            // @ts-expect-error is this a new expected error?
+            readRequests[path].resolve(error({ error: UNEXPECTED_ERROR, message: 'Closed' }));
```

Comment:
cc @mroz22

Reply:
maybe it should be `INTERFACE_UNABLE_TO_OPEN_DEVICE`. this is what you would get over USB when you fail to open device before reading.

Reply:
well, not quite. we are already in opened state and reading, but closeDevice (and now im starting to realize that also disconnect and interface read errors) should interrupt pending read promise

Reply:
INTERFACE_DATA_TRANSFER it is, resolved irl